pkg/cache/memorycache/ttl_cache_test.go

```diff
@@ -19,22 +19,36 @@ import (
 	"testing"
 	"time"
 
+	"github.com/pipe-cd/pipecd/pkg/cache"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-
-	"github.com/pipe-cd/pipecd/pkg/cache"
 )
 
 func TestTTLCache(t *testing.T) {
-	c := NewTTLCache(context.TODO(), 0, 5*time.Second)
-	err := c.Put("key-1", "value-1")
-	require.NoError(t, err)
-	value, err := c.Get("key-1")
-	require.NoError(t, err)
-	assert.Equal(t, "value-1", value)
+	t.Run("test context cancel", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		c := NewTTLCache(ctx, 0, 5*time.Second)
+		err := c.Put("key-1", "value-1")
+		require.NoError(t, err)
+		cancel()
+		err = c.Put("key-2", "value-2")
+		require.NoError(t, err)
+		<-time.After(6 * time.Second)
+		value, err := c.Get("key-2")
+		require.NoError(t, err)
+		assert.Equal(t, "value-2", value)
+	})
```

Comment:
Thank you for adding a test case :) 

[imo] It would be better to describe what is the behavior of the test case. 👀 
Also, it would be enough just to add one key-value pair and check it. :)

```suggestion
	t.Run("remain a value after context canceled", func(t *testing.T) {
		ctx, cancel := context.WithCancel(context.Background())
		c := NewTTLCache(ctx, 0, 5*time.Second)
		err := c.Put("key-1", "value-1")
		require.NoError(t, err)
		cancel()
		<-time.After(6 * time.Second)
		value, err := c.Get("key-1")
		require.NoError(t, err)
		assert.Equal(t, "value-1", value)
	})
```